File: tests/fixtures/skeleton/sample.py
"""Sample Python fixture for Skeletonizer coverage tests."""
import json
from typing import Optional


class UserService:
    def __init__(self, db_path: str) -> None:
        self.db_path = db_path

    def get_user(self, user_id: str) -> Optional[dict]:
        with open(self.db_path, "r") as f:
            data = json.load(f)
        # BODY_SENTINEL_DO_NOT_LEAK
        return data.get(user_id)


def format_user(user: dict) -> str:
    # BODY_SENTINEL_DO_NOT_LEAK
    return f"{user['name']} <{user['email']}>"

File: tests/fixtures/skeleton/sample.go
// Sample Go fixture for Skeletonizer coverage tests.
package sample

import (
	"encoding/json"
	"os"
)

type User struct {
	ID    string
	Name  string
	Email string
}

type UserService struct {
	dbPath string
}

func NewUserService(dbPath string) *UserService {
	return &UserService{dbPath: dbPath}
}

func (s *UserService) GetUser(id string) (*User, error) {
	data, err := os.ReadFile(s.dbPath)
	if err != nil {
		return nil, err
	}
	// BODY_SENTINEL_DO_NOT_LEAK
	var u User
	if err := json.Unmarshal(data, &u); err != nil {
		return nil, err
	}
	return &u, nil
}

func FormatUser(u *User) string {
	// BODY_SENTINEL_DO_NOT_LEAK
	return u.Name + " <" + u.Email + ">"
}
